Add registry to custom FieldTemplate

stanlemon opened the pull request:
### Reasons for making this change

Certain customizations one might want to do in a FieldTemplate may require access to things that are only contained into the registry, such as the rootSchema. Note that this is already done in the ObjectFieldTemplate and so this merely brings the properties that are passed down into alignment with that template.

### Checklist

* [X] **I'm updating documentation**
  - [X] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [X] **I'm adding or updating code**
  - [X] I've added and/or updated tests
  - [X] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

stanlemon commented:
@epicfaace I think I did that correctly, please let me know if I did not. I noticed that `ObjectFieldTemplate` does not include `registry` in the docs or type definition, though it does receive it. I'm happy to modify both of those, but wanted to check if you'd prefer that be in a separate PR?

epicfaace commented:
> @epicfaace I think I did that correctly, please let me know if I did not. I noticed that `ObjectFieldTemplate` does not include `registry` in the docs or type definition, though it does receive it. I'm happy to modify both of those, but wanted to check if you'd prefer that be in a separate PR?

Yeah, feel free to do it in this PR too. Thanks!

stanlemon commented:
Updated `ObjectFieldTemplate` props and updated `ObjectFieldTemplate` and `ArrayFieldTemplate` docs. Re-verified docs looked correct and re-ran tests, all looked 👍 .

stanlemon commented:
Thank you!
